openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -601,14 +601,15 @@ def password_change_request_handler(request):
 
     """
     user = request.user
-    if (user.is_staff or user.is_superuser) and request.POST.get('email_from_support_tools'):
+    request_from_support_tools = (user.is_staff or user.is_superuser) and request.POST.get('email_from_support_tools')
+    if request_from_support_tools:
         email = request.POST.get('email_from_support_tools')
     else:
         # Prefer logged-in user's email
         email = user.email if user.is_authenticated else request.POST.get('email')
     AUDIT_LOG.info("Password reset initiated for email %s.", email)
 
-    if getattr(request, 'limited', False):
+    if getattr(request, 'limited', False) and not request_from_support_tools:
```

Comment:
I am not sure if this is the only place that needs to be changed, can you add some context on why will this work along with unit tests?